### ISSUE TITLE
Change default divider settings according to new guidelines?

### ISSUE THREAD

**Issue by DreaminginCodeZH** [Gericop/Android-Support-Preference-V7-Fix#98]
Material Design Guidelines has been updated with how dividers should be used: https://material.io/guidelines/patterns/settings.html .

> Divider lines
>
> Use divider lines to group together several related settings. Avoid using dividers between each individual setting.

**Comment by Gericop** [comment]
Should this be the default or introduced as a new flag (e.g. `DIVIDER_MATERIAL`)? The former might mess up already existing apps' design if the creators don't want to embrace the material guides.

**Comment by DreaminginCodeZH** [comment]
Maybe mark it as a breaking change? Android 8.0 settings is doing this and Material Design should be the default.

**Comment by Gericop** [comment]
I implemented it as the default divider while keeping the original one as `DIVIDER_OFFICIAL`. It will be released in the next version. One interesting thing is the size of the category items. Currently they have  a 16dp bottom margin by default as this is set by the layout of the support lib. The guide, however, seems to be operating with a 0dp bottom margin so the items seem smaller.

**Comment by DreaminginCodeZH** [comment]
I believe it is the divider that has the 16dp vertical padding and the category has none, which can be inferred from the following image:

![patterns-settings-languages](https://user-images.githubusercontent.com/4469895/31572640-8b3f8818-b05f-11e7-91d8-5a47439799d6.png)

**Comment by Gericop** [comment]
Available since v**26.1.0.3**. You must use `PreferenceFragmentCompatDividers` to apply the MD style automatically.

**Comment by DreaminginCodeZH** [comment]
Hi I updated to 26.1.0.3 and found that the padding before divider is missing, which makes different sections less noticeable and is not conforming to the guidelines. You can checkout out the image I posted in the previous comment of this issue.

**Comment by Gericop** [comment]
Yeah, I left it out... Whoops... However, I started looking at the divider implementation to see how could it be fit in there and just found out that the `getItemOffsets(...)` (that could be used for this padding) doesn't work reliably. The problem is with the item view type determination:

```java
int index = parent.indexOfChild(view);
if (index < parent.getChildCount() - 1) {
    // ...
}
```

This condition seems to be false most of the time because this method is called on-the-fly when adding the items to the layout, so `getChildCount()` doesn't return the proper value, so I'll need to look for a more reliable way to determine the current and next item view type.

_Fun fact: the official lib uses this approach too, so technically it doesn't work there either..._

**Comment by Gericop** [comment]
Is this more like it?

![screenshot_1509052751](https://user-images.githubusercontent.com/5181621/32077600-4355772a-baa4-11e7-9bd2-7fa7af63460c.png)

**Comment by DreaminginCodeZH** [comment]
Yes I think so. Great work!

**Comment by Gericop** [comment]
Yeah, now I only need to determine when to apply this because the category has a bottom margin of 16dp and a top padding of 16dp by default. So I think this kind of padding removal will only be applied if the divider flags are set to be the default one (the material styled), otherwise it will keep everything intact.

**Comment by Gericop** [comment]
Funny thing is that the settings in API 26 (and even 27) does not look like the picture you attached. The paddings are more consistent with the current than the material one. I think it should be a switch after all otherwise people will complain that it doesn't look like the settings app of the official ROM...

**Comment by Gericop** [comment]
I'll need to investigate this further, so it will not be released in 27.0.0.0, but 27.0.0.1 later.
